### PR TITLE
add grpcbox instrumentation library

### PR DESCRIPTION
**Pull request by tsloughter**

### CODE REVIEW

File: instrumentation/opentelemetry_grpcbox/README.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+#{service_protos => [route_guide_pb],
+  unary_interceptor =>
+    grpcbox_chain_interceptor:unary([..., 
+                                     fun otel_grpcbox_interceptor:unary/4, 
+                                     ...])}
+```
+
+### Client
+
+Example in `sys.config` for setting up a
```
Comment (ferd) [inline instrumentation/opentelemetry_grpcbox/README.md:29]: incomplete sentence.

File: instrumentation/opentelemetry_grpcbox/src/opentelemetry_grpcbox.app.src [hunk 2]
```diff
@@ -0,0 +1,14 @@
+{application, opentelemetry_grpcbox,
+ [{description, "grpcbox interceptor for OpenTelemetry instrumentation"},
+  {vsn, "0.1.0"},
+  {registered, []},
+  {applications,
+   [kernel,
+    stdlib
```
Comment (ferd) [inline instrumentation/opentelemetry_grpcbox/src/opentelemetry_grpcbox.app.src:7]: I believe you'd want to depend on grpcbox and the OTel API here.

File: instrumentation/opentelemetry_grpcbox/src/opentelemetry_grpcbox.app.src [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+   ]},
+  {env,[]},
+  {modules, []},
+
+  {licenses, ["Apache 2.0"]},
```
Comment (ferd) [inline instrumentation/opentelemetry_grpcbox/src/opentelemetry_grpcbox.app.src:12]:
```suggestion
  {licenses, ["Apache-2.0"]},
```
I'm not 100% sure but I think there's a demand to be using no whitespace in hex now.

**Review by ferd — CHANGES_REQUESTED** [review]
Minor changes needed I think.

File: instrumentation/opentelemetry_grpcbox/src/opentelemetry_grpcbox.app.src (hunk 3 above)
Comment (tsloughter) [reply instrumentation/opentelemetry_grpcbox/src/opentelemetry_grpcbox.app.src:12]: Yup, thanks, I copied this app from an older app -- rebar3 templates have been updated -- which is why numerous things in the .app are wrong.

**Review by ferd — APPROVED** [review]

**Pull request merged**